### DISCUSSION TITLE
Transparent background in nixvim

### DISCUSSION THREAD

**Discussion by butzist** [nix-community/stylix#459] [Q&A] [answered]
What is the recommended configuration for setting a transparent background in neovim? The theme set by stylix seems to have a solid background despite the opacity settings. Is there a way to achieve this with stylix or do I need to use a plugin like [transparent.nvim](https://github.com/xiyaowong/transparent.nvim)?

**Comment by danth** [comment, marked as answer]
It can be done quite easily by placing the following in your vimrc.

```
autocmd VimEnter * highlight Normal guibg=NONE ctermbg=NONE
```

This may be something that should be built in to Stylix, since we have a similar behaviour with Helix where the background becomes transparent if the terminal opacity setting is < 1.

Not having it transparent in all cases is useful because a terminal which we don't support may have a different colored background.

**Reply by trueNAHO** [reply]
> This may be something that should be built in to Stylix, since we have a similar behaviour with Helix where the background becomes transparent if the terminal opacity setting is < 1.

For reference, the following NixVim declarations are transparency related:

https://github.com/danth/stylix/blob/1ff9d37d27377bfe8994c24a8d6c6c1734ffa116/modules/nixvim/nixvim.nix#L31-L34

> It can be done quite easily by placing the following in your vimrc.
>
> ```
> autocmd VimEnter * highlight Normal guibg=NONE ctermbg=NONE
> ```

Stylix' Neovim support could mimic NixVim's transparency implementation.

Should I open a tracking issue regarding global Stylix transparency support?

**Reply by danth** [reply]
We're probably better off with just having a separate issue for each target which is missing support. No need to keep a tracking issue updated
